tests/e2e/kafkatopic_webhook.go

```diff
@@ -0,0 +1,318 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// TODO(mihalexa): move to k8s.go
+const (
+	dryRunStrategyNone   string = "none"
+	dryRunStrategyClient string = "client"
+	dryRunStrategyServer string = "server"
+)
+
+// TODO(mihalexa): move to k8s.go
+// applyK8sResourceFromTemplateWithDryRun is copy of applyK8sResourceFromTemplate which calls a "--dry-run=<strategy>" kubectl command
+func applyK8sResourceFromTemplateWithDryRun(kubectlOptions k8s.KubectlOptions, templateFile string, values map[string]interface{}, dryRunStrategy string) (string, error) {
+	By(fmt.Sprintf("Generating K8s manifest from template %s for dry-run apply", templateFile))
+	var manifest bytes.Buffer
+	rawTemplate, err := os.ReadFile(templateFile)
+	if err != nil {
+		return "", err
+	}
+	t := template.Must(template.New("template").Funcs(sprig.TxtFuncMap()).Parse(string(rawTemplate)))
+	err = t.Execute(&manifest, values)
+	if err != nil {
+		return "", err
+	}
+
+	By("Replicating terratest's k8s.KubectlApplyFromStringE")
+	tmpfile, err := k8s.StoreConfigToTempFileE(GinkgoT(), manifest.String())
+	if err != nil {
+		return "", err
+	}
+	defer os.Remove(tmpfile)
+
+	return k8s.RunKubectlAndGetOutputE(GinkgoT(), &kubectlOptions, "apply", "-f", tmpfile, "--dry-run="+dryRunStrategy, "--output=yaml")
+}
+
+func testWebhooks() bool {
+	return When("Testing webhooks", func() {
+		// temporary section; to be refactored after kubeconfig injection PR
+		var kubectlOptions k8s.KubectlOptions
+		var err error
+		kubectlOptions, err = kubectlOptionsForCurrentContext()
+		if err != nil {
+			GinkgoT().Fail()
+		}
+
+		testWebhookKafkTopic(kubectlOptions)
+	})
+}
+
+func testWebhookKafkTopic(kubectlOptions k8s.KubectlOptions) {
```

Comment:
Typo
```suggestion
func testWebhookKafkaTopic(kubectlOptions k8s.KubectlOptions) {
```

Reply:
Thank you for catching this. I'll have to update it in the function definition and at the call site.

Reply:
If you are using VSCode/VSCodium the extension CodeSpellChecker (streetsidesoftware.code-spell-checker) can catch this in editing time.